model/sparql/GenericSparql.php

```diff
@@ -1429,6 +1449,35 @@ public function queryLabel($uri, $lang) {
         }
     }
 
+    /**
+     * Query for skos:prefLabels, skos:altLabels and skos:hiddenLabels of a resource.
+     * @param string $uri
+     * @param string $lang
+     * @return array array of prefLabels, altLabels and hiddenLabels - or null if resource doesn't exist
+     */
+    public function queryAllConceptLabels($uri, $lang) {
+        $query = $this->generateAllLabelsQuery($uri, $lang);
+        $result = $this->query($query);
+        $ret = array();
+
+        foreach ($result as $row) {
+            $labelName = $row->prop->localName();
+            if (!in_array($labelName, $ret)) {
```

Comment:
This `if` block may actually be completely redundant, since PHP is quite relaxed about creating new arrays on the fly...so just assigning to `$ret[$labelName][]` (as is done after the `if` block) should be enough, there's no need to create the array explicitly.